### ISSUE TITLE
Multi GPU training for MultiNLI data with MultipleNegativesRankingLoss

### ISSUE THREAD

**Issue by saurabhkumar** [UKPLab/sentence-transformers#2552]
Is there an example or help for using the _fit_ method for multi GPU training when using MultipleNegativesRankingLoss? If not, is there an alternative with another loss function possible for multi GPU training for the MultiNLI data?

**Comment by tomaarsen** [comment]
Hello!

Multi-GPU training has been implemented in this PR: https://github.com/UKPLab/sentence-transformers/pull/2449
In particular, `Example 3` in the first message of that PR should be useful for you. I would recommend running the code with `torchrun --nproc_per_node 4 train.py` where 4 should be updated to your number of GPUs and `train.py` should be the training script.

You can install that branch via:
```
pip install git+https://github.com/tomaarsen/sentence-transformers@v3/trainer_refactor
```
or 
```
git clone https://github.com/tomaarsen/sentence-transformers.git --depth 1 --branch "v3/trainer_refactor"
pip install ./sentence-transformers
```
if the first option doesn't work.

- Tom Aarsen
